src/webots/gui/WbVideoRecorder.cpp

```diff
@@ -56,22 +56,26 @@ class FrameWriterThread : public QThread {
 public:
   FrameWriterThread(unsigned char *frame, int mutexIndex, const QString &fileName, const QSize &resolution, int pixelRatio,
                     int quality, WbView3D *view) :
-    mFrame(frame),
     mMutexIndex(mutexIndex),
     mFileName(fileName),
     mResolution(resolution),
     mPixelRatio(pixelRatio),
     mQuality(quality),
     mView(view),
-    mSuccess(false) {}
+    mSuccess(false) {
+    const int w = mResolution.width() / mPixelRatio;
+    const int h = mResolution.height() / mPixelRatio;
+    mFrame = new unsigned char[4 * w * h];
+    WbView3D::flipAndScaleDownImageBuffer(frame, mFrame, mResolution.width(), mResolution.height(), mPixelRatio);
+  }
+
+  virtual ~FrameWriterThread() { delete[] mFrame; }
 
   void run() override {
     mView->lockPBOMutex(mMutexIndex);
```

Comment:
Is this still required now that the frame is copied and no longer read directly from the PBO?

Reply:
Hum.. good point, probably we can remove it, let me check it.